evm-adapters/src/sputnik/cheatcodes/cheatcode_handler.rs

```diff
@@ -610,6 +610,14 @@ impl<'a, 'b, B: Backend, P: PrecompileSet> CheatcodeStackExecutor<'a, 'b, B, P>
             HEVMCalls::Roll(inner) => {
                 self.add_debug(CheatOp::ROLL);
                 self.state_mut().backend.cheats.block_number = Some(inner.0);
+                // insert a random block hash for the specified block number if it was not
+                // specified already
+                self.state_mut()
+                    .backend
+                    .cheats
+                    .block_hashes
+                    .entry(inner.0)
+                    .or_insert_with(H256::random);
```

Comment:
Why random instead of deterministic, e.g. a hash of the block number? Mainnet block hashes are deterministic in that running tests at a forked block number should mean that tests have that same blockhash every time, so we might want similar behavior here

On that note, it seems blockhash still returns all zeros when running tests against a forked mainnet, but it should return that block's blockhash